src/scaffoldmaker/utils/zinc_utils.py

```diff
@@ -4,10 +4,34 @@
 
 from opencmiss.zinc.context import Context
 from opencmiss.zinc.field import Field
-from opencmiss.zinc.node import Node
+from opencmiss.zinc.node import Node, Nodeset
+from opencmiss.zinc.result import RESULT_OK
 from scaffoldmaker.utils import interpolation as interp
 from scaffoldmaker.utils import vector
 
+class ZincCacheChanges:
+    """
+    Context manager for ensuring beginChange, endChange always called on
+    supplied object, even with exceptions.
+    Usage:
+    with ZincCacheChanges(object):
+        # make multiple changes to object or objects it owns
+    """
+
+    def __init__(self, object):
+        """
+        :param object: Zinc object with beginChange/endChange methods.
+        """
+        self._object = object
+
+    def __enter__(self):
+        self._object.beginChange()
+        return self
+
+    def __exit__(self, *args):
+        self._object.endChange()
+
+
 def getOrCreateCoordinateField(fieldmodule, name='coordinates', componentsCount=3):
```

Comment:
These `getOrCreate*` functions could be canonicalised somewhat.

Reply:
I'm not sure what you mean here?

Reply:
These functions are all pretty much the same there is a lot of code that could be reused.